src/game/editor/editor_history.h

```diff
@@ -0,0 +1,37 @@
+#ifndef GAME_EDITOR_EDITOR_HISTORY_H
+#define GAME_EDITOR_EDITOR_HISTORY_H
+
+#include "editor_action.h"
+
+#include <queue>
+
+class CEditorHistory
+{
+public:
+	CEditorHistory()
+	{
+		m_pEditor = nullptr;
+	}
+
+	~CEditorHistory()
+	{
+		Clear();
+	}
+
+	void RecordAction(std::shared_ptr<IEditorAction> const &pAction);
+	void RecordAction(std::shared_ptr<IEditorAction> const &pAction, const char *pDisplay);
+	void Execute(std::shared_ptr<IEditorAction> const &pAction, const char *pDisplay = nullptr);
+
+	bool Undo();
+	bool Redo();
+
+	void Clear();
+	inline bool CanUndo() const { return !m_vpUndoActions.empty(); }
+	inline bool CanRedo() const { return !m_vpRedoActions.empty(); }
```

Comment:
```suggestion
	bool CanUndo() const { return !m_vpUndoActions.empty(); }
	bool CanRedo() const { return !m_vpRedoActions.empty(); }
```

`Inline` keyword is unnecessary (as long as the method definitions provided *inline*)